nmpolicy/internal/resolver/filter.go

```diff
@@ -32,15 +32,14 @@ var (
 	}
 )
 
-func filter(inputState map[string]interface{}, path ast.VariadicOperator, expectedValue interface{}) (map[string]interface{}, error) {
```

Comment:
What is the motivation to this commit?

Reply:
The filter mechanism should be all at `filter.go` and not run on the top level visitor since it's not related to "replace" or "walk". The motivation is to move the map/slice filtering flags to filter.go and also to do it without keeping state.